Khan/perseus
# Create UI for adding a locked point to interactive graph editor (mafs only)

## Summary:
UI for adding a locked point to interactive graph editor! This will only be for mafs graphs as mafs graphs have a slightly different look and API (and also it was just not working very well with graphie anyway).

This UI includes:
- A dropdown that allows adding elements (just "Point" for now) added to the interactive graph editor.
- Once added, settings for points
  - The ability to delete the setting in question
  - The x and y coordinates of the point
- The removal of the locked point investigation from graph.tsx and related files.

[The ability to change the color of a point](https://khanacademy.atlassian.net/browse/LEMS-1751) is coming in a future PR.

Issue: LEMS-1813

## Test plan:
`yarn jest packages/perseus-editor/src/widgets/__tests__/interactive-graph-editor.test.tsx`
`yarn jest packages/perseus-editor/src/components/__tests__/util.test.ts`

Storybook
- Go to http://localhost:6006/?path=/story/perseus-editor-widgets-interactive-graph-editor--with-locked-points
- Scroll down to the "Add an element" dropdown
- Click "Add an element"
- Click "Point"
- Confirm that a black point appears at (0, 0) on the graph
- Change the coordinates and confirm that the point moves
- Add another point
- Use the trash button to delete a point
- Confirm that the correct point is deleted

## Demo:

https://github.com/Khan/perseus/assets/13231763/9eca725b-651a-4312-b639-bda26163ab8e
